### ISSUE TITLE
[http] requested buffer size 4096 cannot exceedmaximum size

### ISSUE THREAD

**Issue by epcim** [fluent/fluent-bit#419]
What can this error mean, what buffer on what component is by default 4k?

```td-agent-bit[22946]: [2017/11/07 15:27:05] [error] [http] requested buffer size 4096 cannot exceedmaximum size 4096```

input plugins I am using:
```
cpu
disk
kmsg
mem
netif
syslog-3164-local
systemd
tail
```

output is `forward`

**Comment by edsiper** [comment]
@epcim that belongs to a HTTP based output plugin like elasticsearch, influxdb, td, kafka-rest, etc. Please provide a sosreport (fluent-bit -c yourconf.conf --sosreport)

**Comment by epcim** [comment]
well yes, elasticsearch, influxdb are in place.
But I didn't get the point where I may control the buffer size it's logged above.

```
Fluent Bit Enterprise - SOS Report
==================================
The following report aims to be used by Fluent Bit and Fluentd Enterprise
Customers of Treasure Data. For more details visit:

    https://fluentd.treasuredata.com


[Fluent Bit]
    Edition             Community Edition
    Version             0.12.7
    Built Flags          JSMN_PARENT_LINKS JSMN_STRICT FLB_HAVE_TLS FLB_HAVE_SQLDB FLB_HAVE_BUFFERING FLB_HAVE_FLUSH_LIBCO FLB_HAVE_FORK FLB_IS_TD_AGENT FLB_HAVE_PROXY_GO FLB_HAVE_JEMALLOC JEMALLOC_MANGLE FLB_HAVE_REGEX FLB_HAVE_C_TLS FLB_HAVE_SETJMP FLB_HAVE_ACCEPT4 FLB_HAVE_INOTIFY

[Operating System]
    Name                Linux
    Release             3.12.69-60.64.32-default
    Version             #1 SMP Wed Feb 15 08:57:04 UTC 2017 (76392c8)

[Hardware]
    Architecture        x86_64
    Processors          24

[Built Plugins]
    Inputs              cpu mem kmsg tail proc disk netif dummy head health serial stdin tcp mqtt lib forward random syslog
    Filters             grep stdout kubernetes parser record_modifier
    Outputs             counter es exit file forward http influxdb kafka-rest nats null plot stdout retry td lib flowcounter

[SERVER] Runtime configuration
    Flush               10
    Daemon              Off
    Log_Level           Debug

[INPUT] Instance
    Name                tail.0 (tail, id=0)
    Flags               DYN_TAG
    Threaded            No
    Tag                 openstack.nova.consoleauth
    DB                  /var/lib/fluentbit/nova-consoleauth.flb.db
    Buffer_Max_Size     200k
    Chunk_Size          28k
    Exclude_Path        *.gz,*.xz
    Multiline           On
    Multiline_Flush     25
    Parser_Firstline    nova
    Parser_1            nova_1
    Path                /var/log/nova/nova-consoleauth.log*
    Path_Key            path

[INPUT] Instance
    Name                tail.1 (tail, id=1)
    Flags               DYN_TAG
    Threaded            No
    Tag                 openstack.keystone
    DB                  /var/lib/fluentbit/keystone.flb.db
    Buffer_Max_Size     5M
    Chunk_Size          128k
    Exclude_Path        *.gz,*.xz
    Multiline           On
    Multiline_Flush     5
    Parser_Firstline    keystone
    Parser_1            keystone_1
    Path                /var/log/keystone/keystone.log*
    Path_Key            path

[INPUT] Instance
    Name                tail.2 (tail, id=2)
    Flags               DYN_TAG
    Threaded            No
    Tag                 openstack.keystone.manage
    DB                  /var/lib/fluentbit/keystone-manage.flb.db
    Buffer_Max_Size     5M
    Chunk_Size          128k
    Exclude_Path        *.gz,*.xz
    Multiline           On
    Multiline_Flush     5
    Parser_Firstline    keystone
    Parser_1            keystone_1
    Path                /var/log/keystone/keystone-manage.log*
    Path_Key            path

[INPUT] Instance
    Name                tail.3 (tail, id=3)
    Flags               DYN_TAG
    Threaded            No
    Tag                 openstack.neutron.server
    DB                  /var/lib/fluentbit/neutron-server.flb.db
    Buffer_Max_Size     5M
    Chunk_Size          128k
    Exclude_Path        *.gz,*.xz
    Multiline           On
    Multiline_Flush     5
    Parser_Firstline    neutron
    Parser_1            neutron_1
    Path                /var/log/neutron/neutron-server.log*
    Path_Key            path

[INPUT] Instance
    Name                tail.4 (tail, id=4)
    Flags               DYN_TAG
    Threaded            No
    Tag                 openstack.glance.api
    DB                  /var/lib/fluentbit/glance-api.flb.db
    Buffer_Max_Size     5M
    Chunk_Size          128k
    Exclude_Path        *.gz,*.xz
    Multiline           On
    Multiline_Flush     5
    Parser_Firstline    glance
    Parser_1            glance_1
    Path                /var/log/glance/api.log*
    Path_Key            path

[INPUT] Instance
    Name                tail.5 (tail, id=5)
    Flags               DYN_TAG
    Threaded            No
    Tag                 openstack.glance.registry
    DB                  /var/lib/fluentbit/glance-registry.flb.db
    Buffer_Max_Size     5M
    Chunk_Size          128k
    Exclude_Path        *.gz,*.xz
    Multiline           On
    Multiline_Flush     5
    Parser_Firstline    glance
    Parser_1            glance_1
    Path                /var/log/glance/registry.log*
    Path_Key            path

[INPUT] Instance
    Name                tail.6 (tail, id=6)
    Flags               DYN_TAG
    Threaded            No
    Tag                 openstack.glance.scrubber
    DB                  /var/lib/fluentbit/glance-scrubber.flb.db
    Buffer_Max_Size     5M
    Chunk_Size          128k
    Exclude_Path        *.gz,*.xz
    Multiline           On
    Multiline_Flush     5
    Parser_Firstline    glance
    Parser_1            glance_1
    Path                /var/log/glance/scrubber.log*
    Path_Key            path

[INPUT] Instance
    Name                tail.7 (tail, id=7)
    Flags               DYN_TAG
    Threaded            No
    Tag                 openstack.glance.server
    DB                  /var/lib/fluentbit/glance-server.flb.db
    Buffer_Max_Size     5M
    Chunk_Size          128k
    Exclude_Path        *.gz,*.xz
    Multiline           On
    Multiline_Flush     5
    Parser_Firstline    glance
    Parser_1            glance_1
    Path                /var/log/glance/glance.log*
    Path_Key            path

[INPUT] Instance
    Name                tail.8 (tail, id=8)
    Flags               DYN_TAG
    Threaded            No
    Tag                 openstack.glance.manage
    DB                  /var/lib/fluentbit/glance-manage.flb.db
    Buffer_Max_Size     5M
    Chunk_Size          128k
    Exclude_Path        *.gz,*.xz
    Multiline           On
    Multiline_Flush     5
    Parser_Firstline    glance
    Parser_1            glance_1
    Path                /var/log/glance/manage.log*
    Path_Key            path

[INPUT] Instance
    Name                tail.9 (tail, id=9)
    Flags               DYN_TAG
    Threaded            No
    Tag                 openstack.cinder.api
    DB                  /var/lib/fluentbit/cinder-api.flb.db
    Buffer_Max_Size     5M
    Chunk_Size          128k
    Exclude_Path        *.gz,*.xz
    Multiline           On
    Multiline_Flush     5
    Parser_Firstline    cinder
    Parser_1            cinder_1
    Path                /var/log/cinder/api.log*
    Path_Key            path

[INPUT] Instance
    Name                tail.10 (tail, id=10)
    Flags               DYN_TAG
    Threaded            No
    Tag                 openstack.cinder.volume
    DB                  /var/lib/fluentbit/cinder-volume.flb.db
    Buffer_Max_Size     5M
    Chunk_Size          128k
    Exclude_Path        *.gz,*.xz
    Multiline           On
    Multiline_Flush     5
    Parser_Firstline    cinder
    Parser_1            cinder_1
    Path                /var/log/cinder/*volume*
    Path_Key            path

[INPUT] Instance
    Name                tail.11 (tail, id=11)
    Flags               DYN_TAG
    Threaded            No
    Tag                 openstack.cinder.scheduler
    DB                  /var/lib/fluentbit/cinder-scheduler.flb.db
    Buffer_Max_Size     5M
    Chunk_Size          128k
    Exclude_Path        *.gz,*.xz
    Multiline           On
    Multiline_Flush     5
    Parser_Firstline    cinder
    Parser_1            cinder_1
    Path                /var/log/cinder/scheduler.log*
    Path_Key            path

[INPUT] Instance
    Name                tail.12 (tail, id=12)
    Flags               DYN_TAG
    Threaded            No
    Tag                 openstack.cinder.manage
    DB                  /var/lib/fluentbit/cinder-manage.flb.db
    Buffer_Max_Size     5M
    Chunk_Size          128k
    Exclude_Path        *.gz,*.xz
    Multiline           On
    Multiline_Flush     5
    Parser_Firstline    cinder
    Parser_1            cinder_1
    Path                /var/log/cinder/cinder-manage.log*
    Path_Key            path

[INPUT] Instance
    Name                tail.13 (tail, id=13)
    Flags               DYN_TAG
    Threaded            No
    Tag                 pacemaker
    DB                  /var/lib/fluentbit/pacemaker.flb.db
    Buffer_Max_Size     5M
    Chunk_Size          128k
    Exclude_Path        *.gz,*.xz
    Multiline           False
    Multiline_Flush     4
    Parser              pacemaker
    Path                /var/log/pacemaker.log*
    Path_Key            path

[INPUT] Instance
    Name                tail.14 (tail, id=14)
    Flags               DYN_TAG
    Threaded            No
    Tag                 rabbitmq
    DB                  /var/lib/fluentbit/rabbitmq.flb.db
    Buffer_Chunk_Size   128k
    Buffer_Max_Size     5M
    Chunk_Size          128k
    Exclude_Path        *.gz,*.xz
    Multiline           On
    Multiline_Flush     5
    Parser_Firstline    rabbitmq_f
    Parser_1            rabbitmq_1
    Path                /var/log/rabbitmq/rabbit*
    Path_Key            path

[FILTER] Instance
    Name                record_modifie (record_modifier, id=0)
    Match               *
    Record              hostname control-node1.lab42b.zzz.xxx.corp

[OUTPUT] Instance
    Name                forward.0 (forward, mask_id=1)
    Match               openstack.*
    TLS Active          No
    Retry Limit         1
    Host.TCP_Port       30224
    Host.Name           10.20.97.123
    Time_As_Integer     False

[OUTPUT] Instance
    Name                forward.1 (forward, mask_id=2)
    Match               forward.**
    TLS Active          No
    Retry Limit         1
    Host.TCP_Port       30224
    Host.Name           10.20.97.123
    Time_As_Integer     False

[OUTPUT] Instance
    Name                es.0 (es, mask_id=4)
    Match               *
    TLS Active          No
    Retry Limit         no limit
    Host.TCP_Port       30920
    Host.Name           10.20.97.123
    Buffer_Size         1KB
    Include_Tag_Key     On
    Index               fluentbit
    Logstash_Dateformat %Y.%m.%d
    Logstash_Format     Off
    Logstash_Preffix    logstash
    Tag_Key             _tag
    Time_Key            @timestamp
    Time_Key_Format     %Y-%m-%dT%H:%M:%S
    Type                flb_type
```

**Comment by edsiper** [comment]
Fixed by edeab2481153d3b3f68c541c5616236db267db80
